packages/interactivity/src/directives.js

```diff
@@ -102,17 +102,24 @@ export default () => {
 			const inheritedValue = useContext( inheritedContext );
 			const currentValue = useRef( deepSignal( {} ) );
 			const passedValues = context.map( ( { value } ) => value );
-
-			currentValue.current = useMemo( () => {
-				const newValue = context
-					.map( ( c ) => deepSignal( { [ c.namespace ]: c.value } ) )
-					.reduceRight( mergeDeepSignals );
-
-				mergeDeepSignals( newValue, inheritedValue );
-				mergeDeepSignals( currentValue.current, newValue, true );
-				return currentValue.current;
-			}, [ inheritedValue, ...passedValues ] );
-
+			try {
+				const { namespace, value } = context.find(
+					( { suffix } ) => suffix === 'default'
+				);
```

Comment:
I'd say inside in order to prevent conditionals before hooks (although I believe it will work with Preact).

Actually, looking more carefully, I see that `passedValues` is still an array that iterates over all defined `wp-context` directives (line 104), and it's passed inside the `useMemo` dependencies. We can rewrite that part with something like this:

```js
const defaultEntry = context.find( ( { suffix } ) => suffix === 'default' );

currentValue.current = useMemo( () => {
	if ( ! defaultEntry ) return null;

	const { namespace, value } = defaultEntry;
	const newValue = deepSignal( { [ namespace ]: value } );
	mergeDeepSignals( newValue, inheritedValue );
	mergeDeepSignals( currentValue.current, newValue, true );
	return currentValue.current;
}, [ inheritedValue, defaultEntry ] );

if ( currentValue. current ) {
  return <Provider value={ currentValue.current }>{ children }</Provider>
};
```

That way, the `try catch` is not needed.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/58664/commits/b68c44cb56201ff55b958305770104b0c4942e3b